src/pages/profiles/profiles-table.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { RefObject, useCallback, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import {
+    Cancel,
+    CheckCircle,
+    ManageAccounts,
+    RadioButtonUnchecked,
+} from '@mui/icons-material';
+import {
+    GridButton,
+    GridButtonDelete,
+    GridTable,
+    GridTableRef,
+} from '../../components/Grid';
+import { UserAdminSrv, UserProfile } from '../../services';
+import {
+    ColDef,
+    GetRowIdParams,
+    RowDoubleClickedEvent,
+    SelectionChangedEvent,
+    TextFilterParams,
+} from 'ag-grid-community';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+
+const defaultColDef: ColDef<UserProfile> = {
+    editable: false,
+    resizable: true,
+    minWidth: 50,
+    cellRenderer: 'agAnimateSlideCellRenderer',
+    showDisabledCheckboxes: true,
+    rowDrag: false,
+    sortable: true,
+};
+
+export interface ProfilesTableProps {
+    gridRef: RefObject<GridTableRef<UserProfile>>;
+    onRowDoubleClicked: (event: RowDoubleClickedEvent<UserProfile>) => void;
+    setOpenAddProfileDialog: (open: boolean) => void;
+}
+
+const ProfilesTable: React.FunctionComponent<ProfilesTableProps> = (props) => {
+    const intl = useIntl();
+    const { snackError } = useSnackMessage();
+
+    const [rowsSelection, setRowsSelection] = useState<UserProfile[]>([]);
+
+    function getRowId(params: GetRowIdParams<UserProfile>): string {
+        return params.data.id ? params.data.id : '';
+    }
+
+    const onSelectionChanged = useCallback(
+        (event: SelectionChangedEvent<UserProfile, {}>) =>
+            setRowsSelection(event.api.getSelectedRows() ?? []),
+        [setRowsSelection]
+    );
+
+    const onAddButton = useCallback(
+        () => props.setOpenAddProfileDialog(true),
+        [props]
+    );
+
+    const deleteProfiles = useCallback((): Promise<void> | undefined => {
```

Comment:
You don't really need to define return type here

Reply:
yes, done